api/prometheus/v1/api.go

```diff
@@ -75,21 +81,79 @@ type Range struct {
 
 // API provides bindings for Prometheus's v1 API.
 type API interface {
+	// AlertManagers returns an overview of the current state of the Prometheus alert manager discovery.
+	AlertManagers(ctx context.Context) (AlertManagersResult, error)
+	// CleanTombstones removes the deleted data from disk and cleans up the existing tombstones.
+	CleanTombstones(ctx context.Context) error
+	// Config returns the current Prometheus configuration.
+	Config(ctx context.Context) (ConfigResult, error)
+	// DeleteSeries deletes data for a selection of series in a time range.
+	DeleteSeries(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) error
+	// Flags returns the flag values that Prometheus was launched with.
+	Flags(ctx context.Context) (FlagsResult, error)
```

Comment:
It could return a `map[string]string` directly.

Reply:
Done.

Reply:
I don't see the change.

Reply:
@simonpasquier `FlagResult` is now defined as a `map[string]string` whereas before it was defined as a custom struct with the flags data contained inside the `Flags` field.

Not sure what the idiomatic thing to do here is, but figured that the former option and recent change might be more descriptive. I can change it to return `map[string]string` directly if that's more preferable.

Reply:
No strong opinion from my side. Lets get inputs from other reviewers.

Reply:
The reason for using types is that you can manipulate the JSON marshalling. In your case, you should probably add a tag to get a lower-case `flags` in the JSON. See my remark below.